src/Mod/Part/App/Attacher.cpp

```diff
@@ -842,9 +842,19 @@ void AttachEngine::readLinks(const std::vector<App::DocumentObject*>& objs,
 
         auto shape = extractSubShape(objs[i], subs[i]);
         if (shape.isNull()) {
-            FC_THROWM(AttachEngineException,
-                      "AttachEngine3D: null subshape " << objs[i]->getNameInDocument() << '.'
-                                                       << subs[i]);
+            if (subs[i].length() == 0) {
+                storage.emplace_back(TopoShape());
+                shapes[i] = &storage.back();
+                types[i] = eRefType(rtPart | rtFlagHasPlacement);
+                continue;
```

Comment:
There's some lint here, if you feel like it, maybe it could cleaned up at some point @PaddleStroke? :)